### PR TITLE
Support STETH trades that auto wrap to WSTETH

### PR DESCRIPTION
**Pull request by ewilz**

### CODE REVIEW

File: src/entities/protocols/uniswap.ts [hunk 1]
```diff
@@ -43,7 +44,7 @@ export class UniswapTrade implements Command {
   constructor(public trade: RouterTrade<Currency, Currency, TradeType>, public options: SwapOptions) {}
 
   encode(planner: RoutePlanner, _config: TradeConfig): void {
-    let payerIsUser = true
+    let payerIsUser = !this.options.payerIsRouter
```
[the remark this answers is not in the document]
Comment (ewilz) [reply src/entities/protocols/uniswap.ts:47]: correct. I don't think it made any sense for any of the features we had.... wasn't sure if asking for the _opposite_ (payerIsROUTER) as an option was awkward...

File: test/uniswapTrades.test.ts [hunk 2]
```diff
@@ -595,4 +614,157 @@ describe('Uniswap', () => {
...
+      const methodParameters = SwapRouter.swapCallParameters([
+        wrapSTETH,
+        new UniswapTrade(buildTrade([trade]), swapOptions({ payerIsRouter: true })),
+      ])
+      registerFixture('_UNISWAP_V3_001_STETH_FOR_ETH', methodParameters)
+      expect(hexToDecimalString(methodParameters.value)).to.eq('0')
+      // other assertions carried out in forge
+    })
+
+    it('encodes a single exactInput STETH -> WSTETH -> WETH exactOutput swap', async () => {
```
Comment (zhongeric) [inline test/uniswapTrades.test.ts:682]: nit: test name should be exactOut

File: test/uniswapTrades.test.ts [hunk 3]
```diff
@@ -595,4 +614,157 @@ describe('Uniswap', () => {
...
+        ...permit2Data,
+        signature,
+      }
+      const wrapSTETH = new WrapSTETH(inputSTETH, 1, WrapSTETHPermitData)
+      const unwrapSTETH = new UnwrapSTETH(SENDER_AS_RECIPIENT, 0, 1)
+
+      const methodParameters = SwapRouter.swapCallParameters([
+        wrapSTETH,
+        new UniswapTrade(buildTrade([trade]), swapOpts),
+        unwrapSTETH,
```
Comment (zhongeric) [inline test/uniswapTrades.test.ts:708]: do you need an unwrap here for STETH? since its WETH out

File: test/uniswapTrades.test.ts [hunk 4]
```diff
@@ -595,4 +614,157 @@ describe('Uniswap', () => {
...
+      const uniswapTrade = new UniswapTrade(buildTrade([trade]), swapOptions({ recipient: ROUTER_AS_RECIPIENT }))
+      const unwrapSTETH = new UnwrapSTETH(TEST_RECIPIENT_ADDRESS, 1, 1)
+
+      const methodParameters = SwapRouter.swapCallParameters([uniswapTrade, unwrapSTETH])
+      registerFixture('_UNISWAP_V3_001_ETH_FOR_STETH', methodParameters)
+      expect(hexToDecimalString(methodParameters.value)).to.eq(inputETH.toString())
+      // other assertions carried out in forge
+    })
+
+    it('encodes a single exactInput WETH -> WSTETH -> STETH exactOutput swap', async () => {
```
Comment (zhongeric) [inline test/uniswapTrades.test.ts:751]: nit test name

**Review by zhongeric — APPROVED** [review]

File: test/uniswapTrades.test.ts (hunk 3 above)
Comment (ewilz) [reply test/uniswapTrades.test.ts:708]: yes, bc it's exactOut so the input amount is uncertain

**Pull request merged**